synapse/module_api/__init__.py

```diff
@@ -1452,6 +1452,28 @@ async def get_monthly_active_users_by_service(
             start_timestamp, end_timestamp
         )
 
+    async def lookup_room_alias(self, room_alias: str) -> Tuple[str, List[str]]:
+        """
+        Get the room ID associated with a room alias.
+
+        Added in Synapse v1.65.0.
+
+        Args:
+            room_alias: The alias to look up.
+        Returns:
+            A tuple of:
+                The room ID (str).
+                Hosts likely to be participating in the room ([str]).
+        Raises:
+            SynapseError if room alias is invalid or could not be found.
+        """
+        alias = RoomAlias.from_string(room_alias)
+        (room_id, hosts) = await self._hs.get_room_member_handler().lookup_room_alias(
```

Comment:
Is there an issue with getting the room member handler in the `__init__` method (e.g. circular imports)? If not it would be more idiomatic to call `get_room_member_handler` there.

Reply:
> Is there an issue with getting the room member handler in the __init__ method (e.g. circular imports)?

Not from the testing I just did with my (module) use-case. I just followed [your lead](https://github.com/matrix-org/synapse/commit/adc0d35b17952b8b74fbfad663f9bff4e4dd975a#diff-0b449f6f95672437cf04f0b5512572b4a6a729d2759c438b7c206ea249619885R706) :stuck_out_tongue_winking_eye: .

> If not it would be more idiomatic to call get_room_member_handler there.

OK sure, done here: d0dffac